app/controllers/stories_controller.rb

```diff
@@ -335,6 +336,12 @@ def assign_classified_listings
     @classified_listings = ClassifiedListing.where(published: true).select(:title, :category, :classified_listing_category_id, :slug, :bumped_at)
   end
 
+  def redirect_to_lowercase_username
+    return unless params[:username] && params[:username]&.match?(/[A-Z]/)
```

Comment:
even though we only allow letters in this set `[a-zA-Z0-9_]`  for the usernames and slugs, this might change at some point in the future, so to be a little bit more future proofing we might want to use the special `/[[:upper:]]/` that will match any upper case in Unicode. For example:

```ruby
[25] pry(main)> "ŸÊ".match?(/[A-Z]/)
=> false
[26] pry(main)> "ŸÊ".match?(/[[:upper:]]/)
=> true
```

Note that `.upcase` and `.downcase` work correctly:

```ruby
[27] pry(main)> "rhŸmÊs".upcase
=> "RHŸMÊS"
[28] pry(main)> "rhŸmÊs".downcase
=> "rhÿmês"
```

you can find more information about this in https://ruby-doc.org/core-2.7.1/Regexp.html#class-Regexp-label-Character+Classes

Reply:
Wow, TIL about `:upper:` and what it actually does! Surprisingly, it doesn’t show up on Rubular! I think it makes sense to use it and future-proof 👍

Reply:
@rhymes I actually used `/[[:upper:]]/` here initially and then changed it! 😂 That's good to know though! I can make the change back to it using `/[[:upper:]]/` here quickly.